WcaOnRails/app/controllers/results_controller.rb

```diff
@@ -281,8 +281,8 @@ def records
 
     @is_all_years = params[:years] == "all years"
     splitted_years_param = params[:years].split
-    @is_only = splitted_years_param[0] == "only"
-    @is_until = splitted_years_param[0] == "until"
+    @is_only = splitted_years_param[1] == "only"
+    @is_until = splitted_years_param[1] == "until"
```

Comment:
`params[:years]` comes as `"until 2020"`, so `splitted_years_param` is `["until", "2020"]`, thus the current logic is perfectly fine. I actually think the first change you pushed was the right solution (except it should be the other way round as I mentioned) :>

Reply:
Please fetch my branch and try it locally. I believe it's correct as is

Reply:
It certainly doesn't produce an error, but doesn't work either. It basically makes `@is_only` and `@is_until` always false, because their are the result of `"2020" == "only"` and `"2020" == "until"` respectively. If you look one line below you can see that we actually parse `splitted_years_param[1]` as a year.

Looking more on the queries I see the year condition is mostly used where we query the results table, so it doesn't work most of the time (which is the case if you use only/until with results/region or on the records page). I'm surprised this hasn't been noticed sooner actually. Some queries even use the condition in two ways, if you look at [this one](https://github.com/thewca/worldcubeassociation.org/blob/d153d6eb6a765395a176bea4c92329704982a223/WcaOnRails/app/controllers/results_controller.rb#L106-L129), `@years_condition` is first used on the `Concise*Results.year`, but in the outer query it should refer to `Competitions.year`.

All of these queries need to be cleaned up eventually, but I think it's gonna be hard until we can make migrations on the old tables more freely and restructure the db. For now I think the easiest way would be to set the year condition right before each query, so that it uses the right table. I'm sorry you run into this messy part of the codebase, it's been ported quite directly from the PHP version and the query generation is still quite dirty. If you want to try fixing all the year conditions then feel free to do so, otherwise let me know and I can have a closer look on fixing that.